src/main/scala/strawman/collection/Map.scala

```diff
@@ -0,0 +1,44 @@
+package strawman.collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Option
+import scala.annotation.unchecked.uncheckedVariance
+import scala.Predef.???
+
+/** Base Map type */
+trait Map[K, +V]
+  extends Iterable[(K, V)]
```

Comment:
Also, it's hard to see how maps could profit from `Product2`. To gain any efficiency through that, they'd need to already have a Product2 of a key and a value stored somewhere, so they could return that. It's a possibility for mutable hashmaps. Under open hashing, entries could presumably be made into the right Product2 types. But we still don't want to hand out an entry directly, because that could cause a memory leak... So, no.

Regarding "principle of least power": It actually does the other way! Tuple2 is less powerful than Product2 because it constrains the implementation more. So "least power" and "ease of use" are aligned here, after all.